temporalio/client.py

```diff
@@ -5298,50 +5924,117 @@ async def start_workflow_update(
             temporalio.common._apply_headers(
                 input.headers, req.request.input.header.fields
             )
+        return req
 
-        # Repeatedly try to invoke start until the update reaches user-provided
-        # wait stage or is at least ACCEPTED (as of the time of this writing,
-        # the user cannot specify sooner than ACCEPTED)
-        resp: temporalio.api.workflowservice.v1.UpdateWorkflowExecutionResponse
+    async def start_workflow_update_with_start(
+        self, input: StartWorkflowUpdateWithStartInput
+    ) -> WorkflowUpdateHandle[Any]:
+        start_input, update_input = (
+            input.start_workflow_input,
+            input.update_workflow_input,
+        )
+        start_req = (
+            await self._build_update_with_start_start_workflow_execution_request(
+                start_input
+            )
+        )
+        update_req = await self._build_update_workflow_execution_request(
+            update_input, workflow_id=start_input.id
+        )
```

Comment:
I am concerned that exceptions that happen here (e.g. serializing workflow/update args) will leave someone waiting on the workflow handle hanging. Same for things like cancel.

Is it possible to make sure that no matter how this method exits, the start operation handle awaiter is updated? I didn't check if other langs did this, but I think it makes sense.

Reply:
Yes absolutely, good call. Done. Both the exception handling and the inner logic are involved so the inner logic is in a separate function, with a `finally` case in the outer function ensuring that the promise is rejected in all cases.